examples/multigrid-preconditioned-solver/multigrid-preconditioned-solver.cpp

```diff
@@ -75,23 +81,28 @@ int main(int argc, char* argv[])
              }},
             {"dpcpp",
              [] {
-                 return gko::DpcppExecutor::create(0,
-                                                   gko::OmpExecutor::create());
+                 return gko::DpcppExecutor::create(
+                     0, gko::ReferenceExecutor::create());
              }},
             {"reference", [] { return gko::ReferenceExecutor::create(); }}};
 
     // executor where Ginkgo will perform the computation
     const auto exec = exec_map.at(executor_string)();  // throws if not valid
 
+    // const auto fileA = argc >= 3 ? argv[2] : "data/A.mtx";
+    const bool use_mixed = argc >= 4;
+    std::cout << "Use Mixed: " << use_mixed << std::endl;
     // Read data
-    auto A = share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
+    auto A = share(gko::read<mtx>(std::ifstream(argv[2]), exec));
     // Create RHS as 1 and initial guess as 0
     gko::size_type size = A->get_size()[0];
     auto host_x = vec::create(exec->get_master(), gko::dim<2>(size, 1));
-    auto host_b = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    // auto host_b = vec::create(exec->get_master(), gko::dim<2>(size, 1));
+    auto host_b =
+        share(gko::read<vec>(std::ifstream(argv[3]), exec->get_master()));
     for (auto i = 0; i < size; i++) {
         host_x->at(i, 0) = 0.;
-        host_b->at(i, 0) = 1.;
+        // host_b->at(i, 0) = 1.;
```

Comment:
currently working differently from the other example, they probably need to be made to work the same way